# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoaderBase.h
# Commit: 3a34d557d474
# Full Hash: 3a34d557d474adff2245632b993529dd8e01cb3a
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move DOM module loader into own file; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137531
# ==============================================================================

diff -r f29451ea18dc -r 3a34d557d474 dom/script/ModuleLoaderBase.h
--- a/dom/script/ModuleLoaderBase.h	Thu Feb 17 15:57:50 2022 +0000
+++ b/dom/script/ModuleLoaderBase.h	Thu Feb 17 15:57:51 2022 +0000
@@ -86,7 +86,9 @@
       mozilla::MaybeOneOf<JS::SourceText<char16_t>, JS::SourceText<Utf8Unit>>;
 
   // Methods that must be overwritten by an extending class
-  virtual void EnsureModuleHooksInitialized() = 0;
+  virtual void EnsureModuleHooksInitialized() {
+    MOZ_ASSERT(false, "You must override EnsureModuleHooksInitialized");
+  }
   virtual nsresult StartModuleLoad(ScriptLoadRequest* aRequest) = 0;
   virtual void ProcessLoadedModuleTree(ModuleLoadRequest* aRequest) = 0;
   virtual nsresult CompileOrFinishModuleScript(